### PR TITLE
community, docs: update token usage tracking callback + how-to guides

### PR DESCRIPTION
**Pull request by ccurme**

### CODE REVIEW

File: libs/community/langchain_community/callbacks/openai_info.py [hunk 1]
```diff
@@ -210,19 +212,53 @@ def on_llm_new_token(self, token: str, **kwargs: Any) -> None:
...
+            generation = None
+        if isinstance(generation, ChatGeneration):
+            try:
+                message = generation.message
+                if isinstance(message, AIMessage):
+                    usage_metadata = message.usage_metadata
+                else:
+                    usage_metadata = None
+            except AttributeError:
+                usage_metadata = None
```
Comment (ccurme) [inline libs/community/langchain_community/callbacks/openai_info.py:228]: can remove this if we release core and bump in community

File: libs/community/langchain_community/callbacks/openai_info.py [hunk 2]
```diff
@@ -210,19 +212,53 @@ def on_llm_new_token(self, token: str, **kwargs: Any) -> None:
...
+        if usage_metadata:
+            token_usage = {"total_tokens": usage_metadata["total_tokens"]}
+            completion_tokens = usage_metadata["output_tokens"]
+            prompt_tokens = usage_metadata["input_tokens"]
+            if response.llm_output is None:
+                # model name (and therefore cost) is unavailable in
+                # streaming responses
+                model_name = ""
+                if isinstance(generation, ChatGenerationChunk):
+                    warnings.warn("Cost estimates unavailable in streaming mode.")
```
Comment (ccurme) [inline libs/community/langchain_community/callbacks/openai_info.py:240]: will this be noisy?

Comment (baskaryan) [reply libs/community/langchain_community/callbacks/openai_info.py:240]: yea don't think this is needed. also why can't we estimate costs in streaming mode?

**Comment by ccurme** [comment]
cc @eyurtsev @sepiatone

**Pull request merged**